topics/Dialog-and-popup-sizes.md

```diff
@@ -0,0 +1,278 @@
+
+#### Large
+
+Size: 1000×700 px
+
+Best for:
+* Two-column layout featuring components with long names in one of the columns.
+* Three-column layout featuring a table in one of the columns.
+* Four-or-more-column layout possibly featuring a code editor or snippet in any of the columns.
+
+For example, <control>Code Style</control> settings for Java in the <control>Settings</control> dialog. The dialog has three columns with a table and a code snippet in different columns:
+
+![The Code Style settings in the Settings dialog. The dialog has three columns with a table and a code snippet in different columns](dialog_popup_sizes_large.png){width="706"}
+
+### Popup
+
+Popups should have the same default sizes as dialogs, but unlike dialogs, they can have adaptive height. If the popup displays different amounts of content depending on the situation, adapt the popup size to the content on opening.
+
+For example, the height of the <control>Git Brunches</control> popup adapts to the amount of the content inside:
+
+![Two Git Branches popups adapting to the height of their content. The popup on the left is shorter because it has fewer branches, while the popup on the right is longer because it has more branches](dialog_popup_sizes_adaptive.png){width="706"}
+
+When a user changes an adaptive popup’s size, the size is saved and the adaptivity becomes inapplicable.
+
+### Tool windows
+
+* Vertical tool windows should have the default size set to 250×500&nbsp;px.
+* Horizontal tool windows should have the default size set to 500×250&nbsp;px.
+
+The default sizes should be set for all the states of the <control>View Mode</control>.
+
+<table style="none" border="false">
+    <tr>
+        <td width="50%">
+            Vertical <control>Project</control> tool window:
+            <img src="dialog_popup_sizes_tool_window_vertical.png" alt=""/>
+        </td>
+        <td width="50%">
+            Horizontal <control>Build</control> tool window:
+            <img src="dialog_popup_sizes_tool_window_horizontal.png" alt=""/>
+        </td>
+    </tr>
+</table>
+
+### Editor tabs
+
+Place your content into a <control>tab</control> in the editor when there are three or more columns in a row with important information that needs to be shown.
+
+For example, the <control>Merge</control> functionality features a list of changes with actions in a tool window and two code editors with line numbers:
+
+![The Merge functionality with a list of changes with actions in a tool window on the left and two code editors with line numbers in the editor tab on the right](dialog_popup_sizes_editor.png){width="706"}
+
+## Minimum sizes for components
+
+When the default sizes are either too big or too small for a window, or the window has a complex layout, add minimum sizes to important content inside the window. This will make the content displayed correctly and reduce any potential issues caused by resizing the window.
+
+### How to set the size correctly
+
```

Comment:
Possibly add illustrations for types of content here? Not always clear what it is, and when it's clear, an illustration can help visualize the sizes